### PR TITLE
Deprecate `sunpy.physics.solar_rotation`

### LINKED ISSUE #5965: What to do about `physics.solar_rotation`
### Provide a general description of the issue or problem.

When deprecating `image.coalignment` (#5957), it was revealed that `physics.solar_rotation.mapsequence_solar_derotate` depends on `image.coalignment.apply_shifts`. This led me down the rabbit hole of the `physics.solar_rotation` module.

`physics.solar_rotation` contains two functions:

* `calculate_solar_rotate_shift`
* `mapsequence_solar_derotate`

The latter is just a light wrapper around `apply_shifts` (which as mentioned above, we would like to deprecate). The former is just used by the latter to compute the shifts associated with rotation relative to some selected layer.  

I've mainly created this issue to discuss what we should do with this module. Leave it alone? Deprecate it and move it to `sunkit-image`? 

Given that it is such a lightweight wrapper, I could also imagine just adding an example gallery entry in `sunkit-image` on how to use `solar_rotate_coordinate` with `apply_shifts` to shift a map according to some relative shift in the center coordinate of the map. This would also make it explicitly to users what is happening under the hood, i.e. there is no warping or rotation, just shifts in the image plane.

### PR DESCRIPTION
**Pull request by wtbarnes**
## PR Description

Closes #5965 by deprecating `calculate_solar_rotate_shift` and `mapsequence_solar_derotate` and moving them to `sunkit_image.coalignment`.

## PR Checklist

- [ ] I have followed the guidelines in the [Contributing document](https://docs.sunpy.org/en/latest/dev_guide/contents/newcomers.html)
- [ ] Changes follow the coding style of this project
- [ ] Changes have been formatted and linted
- [ ] Changes pass pytest style unit tests (and `pytest` passes).
- [ ] Changes include any required corresponding changes to the documentation
- [ ] Changes have been rebased and squashed to the minimal number (typically 1) of relevant commits
- [ ] Changes have a descriptive commit message with a short title
- [ ] I have added a `Fixes #XXXX -` or `Closes #XXXX -` comment to auto-close the issue that your PR addresses

### CODE REVIEW

**Review by nabobalis — APPROVED** [review]

**Comment by nabobalis** [comment]
I did a release of sunkit-image last night.

**Comment by wafels** [comment]
Fine by me.

**Review by wafels — DISMISSED** [review]

File: sunpy/physics/tests/test_solar_rotation.py [hunk 1]
```diff
@@ -16,6 +16,9 @@
 import sunpy.map
 from sunpy.physics.solar_rotation import calculate_solar_rotate_shift, mapsequence_solar_derotate
 
+DEP_WARNING = r'''ignore:The {} function is deprecated and may be removed in version 4.1.
```
Comment (nabobalis) [inline sunpy/physics/tests/test_solar_rotation.py:19]:
```suggestion
DEP_WARNING = r'''ignore:The {} function is deprecated and will be removed in version 4.1.
```

File: sunpy/physics/tests/test_solar_rotation.py [hunk 2]
```diff
@@ -16,6 +16,9 @@
...
+\s+Use `sunkit_image.coalignment.{}` instead.:sunpy.util.exceptions.SunpyDeprecationWarning'''
```
[the remark this answers is not in the document]
Comment (nabobalis) [reply sunpy/physics/tests/test_solar_rotation.py:20]: Fixed.

[the remark this answers is not in the document]
Comment (wtbarnes) [reply sunpy/physics/tests/test_solar_rotation.py:20]: I would rather not change the wording here. This is what our deprecated decorator emits as well. If we're going to change this wording, it needs to be changed there as well.

[the remark this answers is not in the document]
Comment (nabobalis) [reply sunpy/physics/tests/test_solar_rotation.py:20]: I already did that.

[the remark this answers is not in the document]
Comment (nabobalis) [reply sunpy/physics/tests/test_solar_rotation.py:20]: But wrongly so it didnt work.

[the remark this answers is not in the document]
Comment (nabobalis) [reply sunpy/physics/tests/test_solar_rotation.py:20]: So I am removing the last commit.

**Review by nabobalis — APPROVED** [review]

**Comment by nabobalis** [comment]
**CHECK THE DOC BUILD BEFORE MERGE**

**Pull request merged**